lib/reindex/s3_bucketd.py

```diff
@@ -204,9 +229,11 @@ def get_next_marker(p):
 
             if buckets:
                 yield buckets
-                if name is not None:
-                    # Break on the first matching bucket if a name is given
-                    break
+                if names:
+                    seen_buckets.update(b.name for b in buckets)
```

Comment:
I would rather do the `update` with the bucket name just processed after `buckets.append(bucket)`